include/superior_mysqlpp/utils.hpp

```diff
@@ -75,4 +75,119 @@ namespace SuperiorMySqlpp
     {
         return streamify(streamify(stream, arg), std::forward<Args>(args)...);
     }
+
+    namespace detail
+    {
+        /**
+         * Detects, if all template parameters are references
+         * 
+         * @tparam Args... Types to be checked
+         */
+        template<typename... Args>
+        struct are_arguments_lrefs : std::false_type {};
+
+        /**
+         * Detects, if all template parameters are references
+         *
+         * @tparam Args... Types to be checked
+         */
+        template<typename... Args>
+        struct are_arguments_lrefs<Args&...> : std::true_type {};
+
+        /**
+         * @brief Generic function info obtained from FunctionInfo trait
+         * 
+         * @tparam ResultType type returned by function
+         * @tparam IsConst true_type, if function has been marked const
+         * @tparam IsVolatile true_type, if function has been marked volatile
+         * @tparam Args function arguments
+         */
+        template<typename ResultType, typename IsConst, typename IsVolatile, typename... Args>
+        struct FunctionInfoImpl
```

Comment:
This seems like a useful general purpose trait. As such, I'd rather seen him _minimal_ - destructuring the function type into arguments, result type and qualifiers, but nothing more.
Specifically - raw_arguments  + arguments_are_lvalue_references  would work better as separate traits.
That way is more extensible and you won't pay for stuff you don't need in instantiation time.